ouroboros-consensus-diffusion/test/consensus-test/Test/Consensus/PeerSimulator/Handlers.hs

```diff
@@ -144,3 +146,49 @@ handlerRequestNext currentIntersection blockTree points =
     TipPoint tip' = tip points
 
     trace = tell . pure
+
+handlerBlockFetch ::
+  forall m .
+  IOLike m =>
+  BlockTree TestBlock ->
+  ChainRange (Point TestBlock) ->
+  AdvertisedPoints ->
+  STM m (Maybe BlockFetch, [String])
+handlerBlockFetch blockTree (ChainRange from to) points =
+  runWriterT $ case points of
+    AdvertisedPoints {header = HeaderPoint (At h), block = BlockPoint (At b)} -> do
+      case BT.findFragment (headerPoint h) blockTree of
+        Just f  -> withHpFragment (f, b)
+        Nothing -> error "header point is not in the block tree"
+    AdvertisedPoints {} -> pure Nothing
+  where
+
+    withHpFragment (hpChain, bp) =
+      -- First, check whether the block point is on the same chain, and before or equal to, the header point.
+      case AF.rollback (blockPoint bp) hpChain of
+        Nothing ->
+          -- REVIEW: Should this be fatal?
+          error "Block point isn't on the same chain as header point"
+          -- pure (Just NoBlocks)
```

Comment:
I think it should and we should document somewhere that this is an invariant of point schedules.

Reply:
this was a misconception; it has been removed.